src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/ace/CodeModel.java

```diff
@@ -122,4 +122,8 @@ public native final void insertRoxygenSkeleton() /*-{
       this.insertRoxygenSkeleton && this.insertRoxygenSkeleton();
    }-*/;
    
+   public native final int buildScopeTreeUpToRow(int row) /*-{
+      return this.$buildScopeTreeUpToRow && this.$buildScopeTreeUpToRow(row);
```

Comment:
If the cost is really 1-2 ms per line should we take the chunk size down to
more like 100 lines?
On Fri, Feb 19, 2016 at 12:12 PM Jonathan <notifications@github.com> wrote:

> In
> src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/ace/CodeModel.java
> <https://github.com/rstudio/rstudio/pull/595#discussion_r53503779>:
>
> > @@ -122,4 +122,8 @@ public native final void insertRoxygenSkeleton() /*-{
> >        this.insertRoxygenSkeleton && this.insertRoxygenSkeleton();
> >     }-*/;
> >
> > +   public native final int buildScopeTreeUpToRow(int row) /*-{
> > +      return this.$buildScopeTreeUpToRow && this.$buildScopeTreeUpToRow(row);
>
> When might we not have this.$buildScopeTreeUpToRow ? If we don't have it
> it looks like this will return undefined (so it won't throw right away
> but might later when we access row_).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rstudio/rstudio/pull/595/files#r53503779>.
>